zadania/Program.cs

```diff
@@ -0,0 +1,155 @@
+    distribution[0] = hubs[0].Item1;
+    for (int i = 1; i < masterRandomNumber; i++)
+    {
+        distribution[i]= hubs[i].Item1 + distribution[i-1];
+    }
+    return distribution;
+}
+List<Tuple<int,int,int,int>> generateLinks(List<Tuple<double,int,int>> hubsDensity, List<Tuple<double,int,int>> hubsDesirability, double[] distributionDensity, double[] distributionDesirability, Random ran, int n)
+{
+    List<Tuple<int, int, int, int>> links = new List<Tuple<int, int, int, int>>();
+    for (int i = 0; i < n; i++)
+    {
+        Tuple<int,int> linksA = generateLink(hubsDensity, distributionDensity, ran);
+        Tuple<int,int> linksB = generateLink(hubsDesirability, distributionDesirability, ran);
+        if (!(linksA.Item1 == linksB.Item1 && linksA.Item2 == linksB.Item2))
+        {
+            links.Add(new Tuple<int, int, int, int>(linksA.Item1, linksA.Item2, linksB.Item1, linksB.Item2));
+        }
+    }
+    return links;
+}
+Tuple<int,int> generateLink(List<Tuple<double, int, int>> hubs, double[] distribution, Random ran)
+{
+    double maxDistro = generateMaxDistro(distribution);
+    double randomNumber = Math.Round(ran.NextDouble()*maxDistro,3);
+    int index = binarySearch(distribution, 0, distribution.Length - 1, randomNumber);
+    return new Tuple<int, int>(hubs[index].Item2, hubs[index].Item3);
+}
+double generateMaxDistro(double[] distribution)
+{
+    double distro = 0;
+    for (int i = 0; i<distribution.Length; i++)
+    {
+        if (distribution[i] > distro) { distro = distribution[i]; }
+    }
+    return distro;
+}
+int binarySearch(double[] distribution, int left, int right, double numberToSearchFor)
+{
+    int mid = 0;
+    while(left<right) { 
+        mid = (left + right)/2;
+        if (distribution[mid] > numberToSearchFor)
+        {
+            right = mid;
+        }
+        else
+        {
+            if (distribution[mid + 1] > numberToSearchFor)
+            {
+                return mid;
+            }
+            else
+            {
+                left = mid;
+            }
+        }
+    }
+    return mid;
+}
+void summary(List<Tuple<int,int,int,int>> links)
+{
+    Dictionary<string, int> summary = new Dictionary<string, int>();
+    for (int i = 0; i < links.Count; i++)
+    {
+        string travel = $"z {links[i].Item1},{links[i].Item2} do {links[i].Item3},{links[i].Item4}";
+        if (summary.ContainsKey(travel))
+        {
+            summary[travel]++;
+        }
+        else
+        {
+            summary[travel] = 1;
+        }
+    }
+    foreach (KeyValuePair<string,int> kvp in summary)
+    {
+        Console.WriteLine("Podróż {0} odbyła się {1} razy", kvp.Key, kvp.Value);
+    }
+}
+double[,] densityTable = generateTableDensity(width, height);
+double[,] desirabilityTable = generateTableDesirability(width, height);
+Random rand = new Random();
+int masterRandomNumber = rand.Next(0, Int32.Parse((5 * Math.Sqrt(height * width)).ToString()));
```

Comment:
Za bardzo kombinujesz, `(int)(5 * Math.Sqrt(height * width)`. Tak w ogóle to co chcesz osiągnąć tą linijką? Nazwa nic mi nie mówi